src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1024,16 +1024,24 @@ private function registerPropertyInfoConfiguration(array $config, ContainerBuild
     private function registerCacheConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
     {
         if (!empty($config['pools'])) {
-            $loader->load('cache_adapters.xml');
+            $loader->load('cache_pools.xml');
         }
 
         foreach ($config['pools'] as $name => $poolConfig) {
-            $poolDefinition = new DefinitionDecorator('cache.adapter.'.$poolConfig['type']);
-            $poolDefinition->replaceArgument(1, $poolConfig['default_lifetime']);
+            $type = $poolConfig['type'];
+            if (!isset($config['pools'][$type]) && !$container->has('cache.pool.'.$type)) {
```

Comment:
Isn't the container->has check enough to allow bundles to define their own reusable pools, as services?